Add ability to reemit all of an agent's events from the UI

elsurudo opened the pull request:
Closes: https://github.com/huginn/huginn/issues/2570

Adds "Re-emit all events" option to `/agents/:agent_id/events` screen:

<img width="176" alt="image" src="https://user-images.githubusercontent.com/50223/62166343-ccee3780-b320-11e9-8a02-2c0701e70faf.png">

dsander commented on app/controllers/agents_controller.rb:
```diff
@@ -75,6 +75,20 @@ def event_descriptions
+
+    # `find_each` orders by PK, so events get re-created in the same order
+    @agent.events.find_each do |event|
+      event.reemit!
```
Is this fast enough when an Agent has a lot of Events? I am wondering how many events can reemitted until the web request times out. It might be worth moving the action into a background job.

elsurudo replied:
I had the same thought, and knew someone might bring it up :) It's correct that in order to be scalable here, this should be moved into a background job.

One other thought I had was that if we bring it out to a job, we should probably pass along a timestamp to the job. My rationale is that you want the action to only affect events created _before_ you pushed that button (in case the job is stuck in a queue). Thoughts?

Honestly these are my first PRs here, and I wasn't sure if there would be even looked at or considered, so great to see this.

I will move this to a background job when I get a chance.

dsander replied:
> One other thought I had was that if we bring it out to a job, we should probably pass along a timestamp to the job. My rationale is that you want the action to only affect events created _before_ you pushed that button (in case the job is stuck in a queue). Thoughts?

That is a good point, especially for busy instances. Would passing in the latest Event id work as well? I'd imagine the query being faster than filtering by the `created_at` timestamp.

elsurudo replied:
You're right, that would be better.

elsurudo commented:
I moved it into a job, and also added the following option as part of the confirmation:

<img width="493" alt="image" src="https://user-images.githubusercontent.com/50223/62239955-2b292200-b3d6-11e9-83e6-3f5c43d9087c.png">

elsurudo commented:
By the way, the reason I needed to change the header stuff is that when the modal was in the `<h2>` tag itself, its content text looked large and terrible. 

Instead, I turned the `<h2>`s (where necessary) into `<span class='h2'>`s, which – while being the same size as `<h2>`s – are inline, which means we can put the mexu _next_ to them without being _within_ them.

dsander commented on app/views/agents/_action_menu.html.erb:
```diff
@@ -86,3 +96,30 @@
+              <span class="glyphicon glyphicon-question-sign hover-help" data-content="If this is checked, the original copies of the events will be deleted once they are re-emitted."></span>
+            </div>
+          <% end %>
+          <%= f.hidden_field :disabled, value: (!agent.disabled).to_s %>
```
✂️ this line?

dsander commented on app/views/agents/_action_menu.html.erb:
```diff
@@ -86,3 +96,30 @@
+            <div class="form-group">
+              <%= check_box_tag :delete_old_events %>
+              <%= label_tag :delete_old_events, 'Delete old events' %>
+              <span class="glyphicon glyphicon-question-sign hover-help" data-content="If this is checked, the original copies of the events will be deleted once they are re-emitted."></span>
```
That is a great idea!

elsurudo commented:
Something went wrong with CI the previous time, but now it's all green. Let's get me back onto mainline :)
